### PR TITLE
prepare_save_trading_states

### PR DESCRIPTION
**Pull request by GuillaumeDSM**

### CODE REVIEW

File: config/__init__.py [hunk 1]
```diff
@@ -257,10 +258,12 @@
...
-CONFIG_DEFAULT_EVALUATOR_FILE = "config/default_evaluator_config.json"
-CONFIG_DEFAULT_TRADING_FILE = "config/default_trading_config.json"
-DEFAULT_CONFIG_FILE = "config/default_config.json"
-LOGGING_CONFIG_FILE = "config/logging_config.ini"
+CONFIG_FOLDER = "config"
+CONFIG_DEFAULT_EVALUATOR_FILE = f"{CONFIG_FOLDER}/default_evaluator_config.json"
+CONFIG_DEFAULT_TRADING_FILE = f"{CONFIG_FOLDER}/default_trading_config.json"
+DEFAULT_CONFIG_FILE = f"{CONFIG_FOLDER}/default_config.json"
+LOGGING_CONFIG_FILE = f"{CONFIG_FOLDER}/logging_config.ini"
+SIMULATOR_STATE_SAVE_FILE = f"{CONFIG_FOLDER}/trading_state_history.json"
```
Comment (Werdor) [inline config/__init__.py]: To be removed ?

File: .gitignore [hunk 2]
```diff
@@ -122,3 +122,5 @@ config\.json
 *.ods
 temp_config\.json
 log
+
+config/trading_state_history\.json
```
Comment (Werdor) [inline .gitignore]: To be removed ?

File: octobot.py [hunk 3]
```diff
@@ -130,7 +130,8 @@ def __init__(self, config):
     async def create_exchange_traders(self, ignore_config=False):
         self.async_loop = asyncio.get_running_loop()
         available_exchanges = ccxt.exchanges
-        for exchange_class_string in self.config[CONFIG_EXCHANGES]:
+        target_exchanges = self.config[CONFIG_EXCHANGES]
+        for exchange_class_string in target_exchanges:
```
Comment (Werdor) [inline octobot.py]: Why ?

Comment (GuillaumeDSM) [reply octobot.py]: useless now

Comment (Werdor) [reply octobot.py]: ok thx :smile:

**Pull request merged**